ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -522,6 +522,9 @@ public void onPartitionBecomeBootstrapFromOffline(String partitionName) {
         // sure old store of this replica is deleted (this store may be created in previous replica addition but failed
         // at some point). Then a brand new store associated with this replica should be created and started.
         if (!addBlobStore(replicaToAdd)) {
+          // We have decreased the available disk space in HelixClusterManager#getDiskForBootstrapReplica. Increase it
+          // back since addition of store failed.
+          replicaToAdd.getDiskId().increaseAvailableSpaceInBytes(replicaToAdd.getCapacityInBytes());
```

Comment:
QQ: it seems like we only decreased the available disk space for FullAuto, but not for semi auto. is this backward compatible?

Reply:
Sure. Updated disk usage when creating a replica in Semi-auto mode too. However, for semi-auto, this value is not used since the 'disk -> replica' mapping is pre-decided during cluster map creation. We only use it in FullAuto since the 'disk -> replica' mapping is done in code.